modules/sdk-coin-sui/src/lib/utils.ts

```diff
@@ -30,6 +30,91 @@ export class Utils implements BaseUtils {
   isValidTransactionId(txId: string): boolean {
     throw new Error('Method not implemented.');
   }
+
+  /**
+   * Checks if raw transaction can be deserialized
+   *
+   * @param {string} rawTransaction - transaction in base64 string format
+   * @returns {boolean} - the validation result
+   */
+  isValidRawTransaction(rawTransaction: string): boolean {
+    try {
+      const data = Buffer.from(rawTransaction, 'hex');
+      const trimmedData = new Uint8Array(data.subarray(TYPE_TAG.length));
+      const deserialized = bcs.de('TransactionData', trimmedData);
+      bcs.ser('TransactionData', deserialized);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   * Check the raw transaction has a valid format in the blockchain context, throw otherwise.
+   *
+   * @param {string} rawTransaction - Transaction in base64 string  format
+   */
+  validateRawTransaction(rawTransaction: string): void {
+    if (!rawTransaction) {
+      throw new ParseTransactionError('Invalid raw transaction: Undefined');
+    }
+    if (!this.isValidRawTransaction(rawTransaction)) {
+      throw new ParseTransactionError('Invalid raw transaction');
+    }
+  }
+
+  /**
+   * Validates addresses to check if all exist and are valid Sui public keys
+   *
+   * @param {string} addresses The address to be validated
+   * @param {string} fieldName Name of the field to validate, its needed to return which field is failing on case of error.
+   */
+  validateAddresses(addresses: string[], fieldName: string): void {
+    for (const address of addresses) {
+      this.validateAddress(address, fieldName);
+    }
+  }
+
+  /**
+   * Validates address to check if it exists and is a valid Sui public key
+   *
+   * @param {string} address The address to be validated
+   * @param {string} fieldName Name of the field to validate, its needed to return which field is failing on case of error.
+   */
+  validateAddress(address: string, fieldName: string): void {
+    if (!address || !this.isValidAddress(address)) {
+      throw new BuildTransactionError(`Invalid or missing ${fieldName}, got: ${address}`);
+    }
+  }
+
+  /** @inheritdoc */
+  isValidAddress(address: string): boolean {
+    return this.isHex(address) && this.getHexByteLength(address) === SUI_ADDRESS_LENGTH;
```

Comment:
@linyah9912 we should have added unit tests for all these util methods you implemented. Let's do that in a follow up.